master-firmware/src/lever/lever.c

```diff
@@ -134,3 +134,12 @@ se2_t lever_deposit(lever_t* lever, se2_t robot_pose)
 
     return blocks_pose;
 }
+
+void lever_tidy(lever_t* lever)
+{
+    lever_lock(&lever->lock);
+    lever_pump_set(lever, LEVER_PUMP_OFF);
+    lever_unlock(&lever->lock);
+
+    lever_retract(lever);
```

Comment:
This should be inside the locked section, I think.

Reply:
`lever_retract` already locks inside https://github.com/cvra/robot-software/blob/master/master-firmware/src/lever/lever.c#L68

Reply:
i know. Still lever_tidy is not atomic in its current config, it could be interrupted and changed between unlock and retract.